Refactor remove duplicacy

abhayrajput87 opened the pull request:
Hello @pbruvoll 
This pull request refactors the code responsible for rendering the sub-page selection component for Tekken  character pages. The duplicated code, previously present in app\routes_mainLayout.t8_.$character.index.tsx, app\routes_mainLayout.t8.$character.antistrat.tsx, and app\routes_mainLayout.t8_.$character.meta.tsx, has been centralized into a new file named app\routes\_mainLayout.t8_.$character.nav.tsx.

pbruvoll commented on app/routes/_mainLayout.t8_.$character._index.tsx:
```diff
@@ -14,6 +14,19 @@ import { filterToDescription, getFilterFromParams } from '~/utils/filterUtils'
+  { link: 'antistrat', value: 'Anti strats' },
+]
+
+//  {/* <nav className="flex gap-3">
```
I guess you dont need the comments anymore?

pbruvoll commented on app/routes/_mainLayout.t8_.$character.nav.tsx:
```diff
@@ -0,0 +1,22 @@
+import { NavLink } from '@remix-run/react'
```
When creating a component in the routes folder, it gets a separate url. So this will render if you go to the url /t8/feng/nav.
Can you move this file from app/routes to app/components? And maybe rename the file to just nav.tsx?

pbruvoll commented on app/routes/_mainLayout.t8_.$character.nav.tsx:
```diff
@@ -0,0 +1,22 @@
+  return (
+    <nav className="flex gap-3">
+      {navData.map((obj, index) => (
+        <NavLink key={index} to={obj.link.toString()}>
```
dont think you need to call to string here since link is already a string?
```suggestion
        <NavLink key={index} to={obj.link}>
```

abhayrajput87 replied:
Thank you for the suggestions! I have made all the necessary changes:
1. I have removed comments from the file.
2. I moved the file from "app/routes" to "app/components" and renamed it to "Nav.tsx."
3. Actually, I had used 'String' in the interface instead of 'string,' so the 'to' was giving a type error because it accepts a string. And now I have also changed that to string.

pbruvoll reviewed (commented):
Thank you for the contribution :) 
I've added a couple of suggestions.

abhayrajput87 commented:
Thank you for the suggestions! I have made all the necessary changes:
1. I have removed comments from the file.
2. I moved the file from "app/routes" to "app/components" and renamed it to "Nav.tsx."
3. Actually, I had used 'String' in the interface instead of 'string,' so the 'to' was giving a type error because it accepts a string. And now I have also changed that to string.

pbruvoll commented:
Thank you, lets get this pr merged :)
